### PR TITLE
fix(rest): Ensure visibility field is case-insensitive

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: Closes : #2569 

### How To Test?
1: URL: http://localhost:8080/resource/api/projects (post)
2: Include the visibility field in the request body.
3: Set the visibility field to both uppercase (e.g., "PRIVATE") and lowercase (e.g., "private").

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -2168,19 +2169,31 @@ private Project convertToProject(Map<String, Object> requestBody) {
...
+                        linkedProjects.put(entry.getKey(), projectRelationshipMap);
+                    }
+                });
+            }
+            JsonNode jsonNode = mapper.valueToTree(requestBody);
+            return mapper.convertValue(requestBody, Project.class);
+        } catch (IllegalArgumentException e) {
+            System.err.println("Error converting requestBody to Project: " + e.getMessage());
+            System.err.println("Failed requestBody: " + requestBody);
+            throw e;
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2195]: Please close the catch section with case changing `if` condition, that is the only modified section of this file. Existing code was not modified in this PR, thus new exception is not expected for entire function.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -2168,19 +2169,31 @@ private Project convertToProject(Map<String, Object> requestBody) {
         ObjectMapper mapper = new ObjectMapper();
         mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         mapper.registerModule(sw360Module);
+        try {
+            if (requestBody.containsKey(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY))) {
+                String visibility = (String) requestBody
+                        .get(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY));
+                requestBody.put(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY),
+                        visibility.toUpperCase());
+            }
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2178]: Close `try/catch` after this `if` statement.

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor change needed.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java (hunk 2 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2183]: comment addressed.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 3]
```diff
@@ -2169,17 +2170,31 @@ private Project convertToProject(Map<String, Object> requestBody) {
...
-                    linkedProjects.put(entry.getKey(), projectProjectRelationShip);
+                    Map<String, Object> projectRelationshipMap = new HashMap<>();
+                    projectRelationshipMap.put("projectRelationship", entry.getValue());
+                    linkedProjects.put(entry.getKey(), projectRelationshipMap);
                 }
             });
-
         }
+
+        JsonNode jsonNode = mapper.valueToTree(requestBody);
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2197]: The variable `jsonNode` is not used anywhere. Is it required?

Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2197]: not required. let me delete this.

Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: done.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**